core/templates/dev/head/domain/exploration/StateObjectFactory.js

```diff
@@ -18,40 +18,26 @@
  */
 
 oppia.factory('StateObjectFactory', [
-  'AnswerGroupObjectFactory',
-  function(AnswerGroupObjectFactory) {
+  'AnswerGroupObjectFactory', 'InteractionObjectFactory',
+  function(AnswerGroupObjectFactory, InteractionObjectFactory) {
   var State = function(name, content, interaction, paramChanges) {
     this.name = name;
+    this.classifier_model_id = null;
     this.content = content;
-    this.interaction = {
-      answer_groups:
-        generateAnswerGroupsFromBackend(interaction.answer_groups),
-      confirmed_unclassified_answers:
-        interaction.confirmed_unclassified_answers,
-      customization_args: interaction.customization_args,
-      default_outcome: interaction.default_outcome,
-      fallbacks: interaction.fallbacks,
-      id: interaction.id
-    };
+    this.interaction = generateInteractionFromBackend(interaction);
     this.paramChanges = paramChanges;
   };
 
-  var generateAnswerGroupsFromBackend = function(answerGroupBackendDicts) {
-    var answerGroups = answerGroupBackendDicts.map(function(answerGroupDict) {
-      return AnswerGroupObjectFactory.create(
-        answerGroupDict.rule_specs,
-        answerGroupDict.outcome,
-        answerGroupDict.correct);
-    });
-
-    return answerGroups;
+  var generateInteractionFromBackend = function(interactionBackendDict) {
```

Comment:
Not sure you really need a separate function for this; you could just inline it.